engine/fasthttp/request.go

```diff
@@ -116,10 +116,25 @@ func (r *Request) FormValue(name string) string {
 // FormParams implements `engine.Request#FormParams` function.
 func (r *Request) FormParams() (params map[string][]string) {
 	params = make(map[string][]string)
-	r.PostArgs().VisitAll(func(k, v []byte) {
-		// TODO: Filling with only first value
-		params[string(k)] = []string{string(v)}
-	})
+	mf, err := r.RequestCtx.MultipartForm()
+
+	if err == fasthttp.ErrNoMultipartForm {
+		r.PostArgs().VisitAll(func(k, v []byte) {
+			// TODO: Filling with only first value
+			params[string(k)] = []string{string(v)}
+		})
+		return
+	}
+
+	if err != nil {
+		panic(err)
```

Comment:
I'm not sure about this

```go
func (req *Request) MultipartForm() (*multipart.Form, error) {
	if req.multipartForm != nil {
		return req.multipartForm, nil
	}

	req.multipartFormBoundary = string(req.Header.MultipartFormBoundary())
	if len(req.multipartFormBoundary) == 0 {
		return nil, ErrNoMultipartForm
	}

	ce := req.Header.peek(strContentEncoding)
	body := req.bodyBytes()
	if bytes.Equal(ce, strGzip) {
		// Do not care about memory usage here.
		var err error
		if body, err = AppendGunzipBytes(nil, body); err != nil {
			return nil, fmt.Errorf("cannot gunzip request body: %s", err)
		}
	} else if len(ce) > 0 {
		return nil, fmt.Errorf("unsupported Content-Encoding: %q", ce)
	}

	f, err := readMultipartForm(bytes.NewReader(body), req.multipartFormBoundary, len(body), len(body))
	if err != nil {
		return nil, err
	}
	req.multipartForm = f
	return f, nil
}
```

Probably the `FormParams` method should return an error if it fails